modules/Contacts/views/view.detail.php

```diff
@@ -53,6 +53,17 @@ class ContactsViewDetail extends ViewDetail
 	public function display(){
 		global $sugar_config;
 
+		// Fix for issue 1506: removed picture is still cached in the Detail View.
```

Comment:
this will only work for contacts and may not be the best approach to solving this issue, I think  a global solution for images is needed